### DISCUSSION TITLE
Questions about effective midi note routing within DAW to Ctrlr

### DISCUSSION THREAD

**Discussion by puffer3** [RomanKubiak/ctrlr#476] [Q&A]
So I have a strange behavior I am trying to understand.  I wrote a panel (with some help from forum friends!) to control a friends Tauntek upgraded Oberheim OB8 synthesizer.  For most of the panels I have been using that I wrote for myself (prophet 600 and OBsX, both soon to be available publicly) I have been using the MID routing method of "Enabling input from plugin host" and then routing using "Host to Out Device"

Then in ableton I set up an additional track for recording audio.  This suited my workflow better than the External Instrument method.

So using the panel in Ableton and blasting the synth with as much note information as I could (32nd note simultaneous chord arpeggiator and a sequence) the synth handles everything fine!  But when I put the panel on the end users computer (a pc running fruity loops) and route the midi the same way as above, I am getting this weird behavior where if I stop playback while a note is on, the note sticks!! and no amount of midi CC 123 panic messages seems to work to turn it off.

On the same computer I programmed an obscene amount of midi to try and make the synth crash without the ctrlr panel and I COULD NOT do it! 

The synth had had some problems sticking that I assumed were because of too much midi information, but I just reseated all the eproms and even bought new ram chips on eBAY which seemed to eliminate hardware malfunction.  I also checked in with Bob at Tauntek to get his thoughts, which were that it should be able to handle it.  I was blasting it with so much midi from my Ableton Mac that I am sure the hardware is ok.  Fruity loops seems ok too and so now I am wondering how I can get the panel to behave better with the midi. Theres not a lot of access to that part of the program so I wanted to ask here if anyone had any advice or experience with these sorts of issues.   Mainly that routing the MIDI thru ctrlr from the host daw  FLstudio on PC seems to be causing problems. 

The interface is a MioXL.

**Comment by damiensellier** [comment]
Sorry but I never used FLstudio and never experienced such problem.

**Comment by Godlike-Productions** [comment]
It's weird that a MIDI Panic won't silence the note.  I'm suspecting you may have a MIDI loop somewhere. Try unplugging the MIDI out from the synth and see if that fixes it, and double check that there isn't a MIDI input into the panel from Fruity Loops that keeps sending MIDI data.

In the Ctrlr hosted panel, you can open up the MIDI monitors and turn on logging for MIDI In and MIDI Out.  That will tell you if you have a MIDI loop.  I'm not familiar with how Fruity manages MIDI routing so this is a bit of a stab in the dark.

You might be able to put add a MIDI in method to the panel that listens for MIDI stop and sends an all notes off message, or there may be an option in Fruity to send note off's when the sequencer is stopped. I know other sequencers have this option.

Also double check the channel assignment.  I notice that a few of your methods have hard coded B0 (cc on Channel 1), even if you set the MIDI channel to something else via the MIDI settings.  The updated method I sent you has the code, for detecting the MIDI channel and using that instead. (For everyone else - see https://github.com/damiensellier/CtrlrX/wiki/Getting-MIDI-Device-and-Channel)

**Reply by puffer3** [reply]
Its not a midi loop as I use a midi monitor when I am sending out notes.  

I have tried sending the panic message other ways ( not ctrlr ) and it still doesnt silence.  I think the tauntek upgrade might not have the midi panic implemented right? 

I will see about the FL midi stop. 

I did do some sloppy coding with hard encoded channels, I will adjust.  Thanks Jason!
